allennlp/common/file_utils.py

```diff
@@ -243,6 +242,46 @@ def _find_latest_cached(url: str, cache_dir: str) -> Optional[str]:
     return None
 
 
+class CacheFile:
```

Comment:
That's some good stuff. I think the logging can be at the `debug` level, but other than that, this is the kind of abstraction we should have used in datastore or beakerstore.

Reply:
I wish it had a better name, because you can use this for many things, not just caches. But `AtomicallyWrittenFile` is awkward, and I can't think of anything better.

Reply:
Yea... I struggled to come up with anything better